hydra-cluster/src/CardanoNode.hs

```diff
@@ -169,9 +178,34 @@ withBlockfrostBackend ::
 withBlockfrostBackend _tracer stateDirectory action = do
   args <- setupCardanoDevnet stateDirectory
   shelleyGenesis <- readFileBS >=> unsafeDecodeJson $ stateDirectory </> nodeShelleyGenesisFile args
-  let backend = BlockfrostBackend $ BlockfrostOptions{projectPath = Backend.blockfrostProjectPath}
+  bfProjectPath <- findFileStartingAtDirectory 3 Backend.blockfrostProjectPath
+  let backend = BlockfrostBackend $ BlockfrostOptions{projectPath = bfProjectPath}
   action (getShelleyGenesisBlockTime shelleyGenesis) backend
 
+-- | Find the given file in the current directory or its parents.
+--
+-- This function starts from the current working directory and checks if the specified file exists there.
+-- If not found, it recursively checks the parent directories up to the given maximum depth.
+findFileStartingAtDirectory :: Int -> FilePath -> IO FilePath
+findFileStartingAtDirectory maxDepth fileName = do
+  cwd <- getCurrentDirectory
+  findInDir maxDepth cwd
+ where
+  findInDir :: Int -> FilePath -> IO FilePath
+  findInDir depth dir = do
+    let path = dir </> fileName
+    exists <- doesFileExist path
+    if exists
+      then pure path
+      else
+        if depth <= 0
+          then error $ "Could not locate the Blockfrost project file at or above: " <> pack dir
```

Comment:
Maybe error here could be more related to the initial input. For example, can not locate in `current directory` and `maxDepth` above. We can also specified the last checked level (`pack dir`). Wdyt?